exporter/dynatraceexporter/metrics_exporter.go

```diff
@@ -275,6 +265,24 @@ func (e *exporter) sendBatch(ctx context.Context, lines []string) error {
 		return consumererror.NewPermanent(fmt.Errorf("metrics ingest v2 module not found - ensure module is enabled and endpoint is correct"))
 	}
 
+	if resp.StatusCode == http.StatusTooManyRequests {
+		return consumererror.NewPermanent(
+			fmt.Errorf("The server responded that too many requests have been sent. Please check your export interval and batch sizes and see https://www.dynatrace.com/support/help/dynatrace-api/basics/access-limit for more information"),
+		)
+	}
+
+	if resp.StatusCode >= http.StatusBadRequest {
+		return consumererror.NewPermanent(fmt.Errorf(`Received error response status: "%v"`, resp.Status))
```

Comment:
Even if only _some_ data points are invalid (e.g. metric key too long or only invalid characters), `400 Bad Request` will be returned, while the valid data points are still accepted:
https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/post-ingest-metrics#response-codes
Would the permanent error only prevent those from being retried or would it disable the entire exporter?

Reply:
A `400` status code returned by Dynatrace receives special handling and wont hit this case, I simply started at 400 to be entirely inclusive of all 4xx+ errors. I can update the starting error code or add a comment if this is unclear.

A permanent error is intended to indicate that the same input will always result in the same output, so the Collector shouldn't try to process the same telemetry data again. A permanent error doesn't disable the exporter, we have to set the `isDisabled` property to disable it, and we have unit tests that ensure we only set it when we intend to.

Reply:
Ah sorry, I only saw that clause above now. I think making it a `>` would make it more obvious (although a comment "400 handled above" in addition wouldn't hurt).

Reply:
```suggestion
	if resp.StatusCode > http.StatusBadRequest { // "400 Bad Request" itself is handled above
		return consumererror.NewPermanent(fmt.Errorf(`Received error response status: "%v"`, resp.Status))
```